packages/runtime/src/runtimes/react/controls/control.tsx

```diff
@@ -47,6 +48,20 @@ type ControlValueProps = {
   control?: InstanceType<ControlDefinition>
 }
 
+type Result<T> = { success: true; value: T } | { success: false; error: string }
+
+export function resolveValue(
+  data: DataType<ControlDefinition> | undefined,
+  definition: ControlDefinition,
+): Result<ResolvedValueType<ControlDefinition>> {
+  try {
+    return { success: true, value: definition.resolveValue(data).readStableValue() }
```

Comment:
Handling errors coming from the unified controls interface only. I originally added an error boundary around `ElementData`'s `<Suspense>` boundary and let these propagate there, but in the `localhost`/dev setup this still results in React showing its (collapsed) errors banner on the page, and that didn't look right to me.

Reply:
Note, React’s error banner even happens on console.error and is controversial in the community. I wouldn’t weigh it heavily

Reply:
> React’s error banner even happens on console.error

I haven't see that.

> I wouldn’t weigh it heavily

I guess it depends on whether we consider this scenario (mismatching prop data) to be worth of polish. For all of its seeming triviality, the React banner really spoils the impression that we have everything under control.